### ISSUE TITLE
Can't Save Lightning Application

### ISSUE THREAD

**Issue by banderson5144** [joeferraro/MavensMate-Desktop#102]
Windows 8.1 (64 bit)
MavensMate-App 0.0.10

Was developing Lightning Component from Atom.io via MavensMate-App.
Then I went through Salesforce Developer Console and created Lightning App.
Went back into MavensMate-App to edit project and bring in Lightning App.
Made change in .app file and tried to save and got this error:

```
> Compiling Ltng_Lookup_Test.app
> Could not update lightning components: Cannot read property 'Id' of undefined
{
  "id": "fc374d70-7d95-11e5-a4f4-ebba03ec3874",
  "complete": true,
  "operation": "compile-metadata",
  "error": "Could not update lightning components: Cannot read property 'Id' of undefined",
  "stack": "Error: Could not update lightning components: Cannot read property 'Id' of undefined\n    at C:\\Program Files\\MavensMate\\resources\\app\\node_modules\\mavensmate\\lib\\mavensmate\\lightning.js:189:18\n    at tryCatch1 (C:\\Program Files\\MavensMate\\resources\\app\\node_modules\\mavensmate\\node_modules\\bluebird\\js\\main\\util.js:45:21)\n    at Promise._callHandler (C:\\Program Files\\MavensMate\\resources\\app\\node_modules\\mavensmate\\node_modules\\bluebird\\js\\main\\promise.js:571:13)\n    at Promise._settlePromiseFromHandler (C:\\Program Files\\MavensMate\\resources\\app\\node_modules\\mavensmate\\node_modules\\bluebird\\js\\main\\promise.js:581:18)\n    at Promise._settlePromiseAt (C:\\Program Files\\MavensMate\\resources\\app\\node_modules\\mavensmate\\node_modules\\bluebird\\js\\main\\promise.js:733:18)\n    at Promise._settlePromises (C:\\Program Files\\MavensMate\\resources\\app\\node_modules\\mavensmate\\node_modules\\bluebird\\js\\main\\promise.js:845:14)\n    at Async._consumeFunctionBuffer (C:\\Program Files\\MavensMate\\resources\\app\\node_modules\\mavensmate\\node_modules\\bluebird\\js\\main\\async.js:85:12)\n    at Async.consumeFunctionBuffer (C:\\Program Files\\MavensMate\\resources\\app\\node_modules\\mavensmate\\node_modules\\bluebird\\js\\main\\async.js:40:14)\n    at doNTCallback0 (node.js:416:9)\n    at process._tickCallback (node.js:345:13)",
  "promiseId": "fc2461b0-7d95-11e5-9356-9d71b2a06c09"
}
```

Looking through the .lightning JSON file, I didn't see my Lightning App listed in there. Figure this could be the problem.

**Comment by dnalot** [comment]
A couple workarounds in joeferraro/MavensMate#569

**Comment by dnalot** [comment]
TLDR; I delete my `config/.lightning` directory every time I create a new JS/CSS in a component bundle.

**Comment by JoernBerkefeld** [comment]
why was this marked as closed?! the bug obviously was NOT fixed. 
The workaround works. Obviously some internals are messed up here.

**Comment by mikepaisner** [comment]
I just ran into this problem.  Deleting the .lightning file in the config folder worked, but there should be a better fix.

**Comment by Lufan** [comment]
Bug is still present in the v0.0.11-beta.7. In .lightning file a record for one component is missed and that error appear when I try to save it.

**Comment by jojoforce** [comment]
Still an issue today. Was there any fix?

**Comment by kidtsunami** [comment]
@JoernBerkefeld because we moved to centralize all of the issues in the main project https://github.com/joeferraro/MavensMate/issues since it was hard to identify duplicate issues as people would create them in what appeared to be random projects.

As such, there is no good secure way to copy all of one repo's issues into another repo in GitHub. We closed all of them as it was going to take forever to manually copy them.

Can you create this issue in the main project and I'll triage it appropriately? (That way I'm not the creator/owner of the issue)
